web3/module.py

```diff
@@ -1,8 +1,38 @@
+from web3.exceptions import (
+    UndefinedMethodError,
+)
+from web3.method import (
+    lookup_method,
+)
+
+
 class Module:
     web3 = None
+    method_class = None
+    module_config = None
+    lookup_method_fn = lookup_method
 
     def __init__(self, web3):
         self.web3 = web3
+        if self.module_config is None:
+            self.module_config = dict()
+
+    def __getattr__(self, attr):
```

Comment:
I'm :-1: on using `__getattr__`.  It makes introspection and type hinting not work well.

I'd advocate for something like:

```python
class Module:
    get_version: GetVersionMethod  # this is how we get strong-ish type hints.

    def __init__(self, ...):
        self.get_version = GetVersion(...)
```

In theory we can do this by assigning them at the class level and using the descriptor API to get access to the `w3` instance.